dynamic_programming/lcs.c

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file
+ * @brief [LCS](https://en.wikipedia.org/wiki/Longest_common_subsequence_problem) algorithm
+ * @details
+ * From Wikipedia: The longest common subsequence (LCS) problem is the problem
+ * of finding the longest subsequence common to all sequences in a set of sequences
+ * (often just two sequences).
+ * @author [Kurtz](https://github.com/itskurtz)
+ * @see dynamic_programming/lcs.c
```

Comment:
`@see` is only used for other similar implementations within the code.
For example, a file named `quick_sort.cpp` mentions the `quick_sort_2.cpp` implementation.